lib/vmdb/initializer.rb

```diff
@@ -7,8 +7,7 @@ def self.init
       #   * command line(rails server)
       #   * debugger
       if defined?(Rails::Server)
-        # preload_for_worker_role depends on seeding, principally MiqDatabase
-        EvmDatabase.seed_primordial
+        raise "expected database to be seeded" unless MiqDatabase.exists?
```

Comment:
@jrafanie would it make more sense to push the seeding check down into `MiqUiWorker.preload_for_worker_role?`? The comment above implies they are intimately connected, so I think it makes more sense there.

Reply:
hmm, it feels like you lose the context and leaves me 😕 if we move it to a method that does this:

```ruby
    def preload_for_worker_role
      configure_secret_token
    end

    def configure_secret_token(token = MiqDatabase.first.session_secret_token)
      return if Rails.application.config.secret_key_base

      Rails.application.config.secret_key_base = token

      # To set a secret token after the Rails.application is initialized,
      # we need to reset the secrets since they are cached:
      # https://github.com/rails/rails/blob/4-2-stable/railties/lib/rails/application.rb#L386-L401
      Rails.application.secrets = nil
    end
```

I'm 50/50 on it.  We could possibly raise from that method `unless MiqDatabase.first?` but it seems to lose the context of seeding.  What do you think?

Reply:
I was thinking of adding a ensure_seeded method or something for readability and so we know that the MiqDatabase check is more about general seeding

Reply:
Maybe we can make EvmDatabase.seeded_primordially? And call it from here.  Current implementation will be MiqDatabase.exists? But in the future it could be changed to something more robust.

Reply:
If you mean to call  `EvmDatabase.seeded_primordially?` from init, I like it.  👍